### PR TITLE
Don't print empty messages

### PR DESCRIPTION
**Pull request by codedmart**
@ocharles I tested this in my project and it seems to work fine. Let me know if I am missing something. Thanks!

### CODE REVIEW

File: src/Control/Monad/Log.hs [hunk 1]
```diff
@@ -491,8 +491,11 @@ withBatchedHandler BatchingOptions{..} flush k =
                        flushAfter flushAlarm <|> flushFull <|> flushOnClose
                      stillOpen <- fmap not (readTVar closed)
                      return (messages,stillOpen))
-             flush messages
-             pure stillOpen
+             case messages of
+                [] -> pure stillOpen
+                msgs ->
+                    do flush msgs
```
Comment (ocharles) [inline src/Control/Monad/Log.hs]: This looks good, but I think the type of `flush` should change. We only ever call ` flush` with a non-empty list, so we should use `Data.List.NonEmpty`.

Comment (codedmart) [reply src/Control/Monad/Log.hs]: @ocharles OK I will take a look and update this. Thanks!

**Comment by codedmart** [comment]
@ocharles OK how about now? Thanks!

**Comment by ocharles** [comment]
Perfect, thank you! Will give it a final test and release tomorrow. :heart:

**Comment by codedmart** [comment]
Just as an FYI I did test on ghc 8 and ghc 7.10. Works great in my project. Thanks!

**Comment by ocharles** [comment]
Released!

**Comment by codedmart** [comment]
Thanks!

**Pull request merged**